Sources/MapboxCoreNavigation/EHorizon/RoadObjectMatcher.swift

```diff
@@ -0,0 +1,113 @@
+import Foundation
+import MapboxNavigationNative
+
+/**
+ Provides methods for road object matching.
+
+ Matching results are delivered asynchronously via a delegate.
+ In case of error (if there are no tiles in the cache, decoding failed, etc.) the object won't be matched.
+ */
+final public class RoadObjectMatcher {
+
+    /// Road object matcher delegate.
+    public weak var delegate: RoadObjectMatcherDelegate? {
+        didSet {
+            if delegate != nil {
+                native.setListenerFor(self)
+            } else {
+                native.setListenerFor(nil)
+            }
+        }
+    }
+
+    /**
+     Matches given OpenLR object to the graph.
+
+     - parameter location: OpenLR location of the road object, encoded in a base64 string.
+     - parameter standard: Standard used to encode OpenLR location.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchOpenLR(location: String, standard: OpenLRStandard, identifier: RoadObjectIdentifier) {
+        native.matchOpenLR(forBase64Encoded: location, standard: MapboxNavigationNative.OpenLRStandard(standard), id: identifier)
+    }
+
+    /**
+     Matches given polyline to the graph.
+     Polyline should define a valid path on the graph,
+     i.e. it should be possible to drive this path according to traffic rules.
+
+     - parameter polyline: Polyline representing the object.
+     - parameter identifier: Unique identifier of the object.
+     */
+    public func matchPolyline(_ polyline: [CLLocation], identifier: RoadObjectIdentifier) {
```

Comment:
For consistency, this method should be called `match(_:identifier:)` and accept a `Turf.LineString`.